heat/core/dndarray.py

```diff
@@ -730,6 +730,32 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
             slices = [slice(None)] * (self.ndim - (len(kst) + len(kend)))
             key = kst + slices + kend
 
+        def __is_singular(obj: any) -> bool:
+            """
+            Checks if obj has size one, and thereby reduces the dimensionalty of
+            the output
+            """
+            try:
+                int(obj)
+                return True
+            except Exception:
+                pass
+            if hasattr(obj, "__len__") and len(obj) == 1:
+                return True
+            if hasattr(obj, "shape") and len(obj.shape) == 0:
+                return True
+            if hasattr(obj, "numel"):
+                return obj.numel() == 1
+            if hasattr(obj, "size"):
+                return obj.size == 1
```

Comment:
probably not, but i wanted to be as duck-typing friendly as possible. anyway, utilizing torch is way better because supporting ducktypes and array-likes not supported by torch would only lead to new errors